### ISSUE TITLE
Improve Security

### ISSUE THREAD

**Issue by julesbravo** [searchspring/magento2-searchspring-feed#41]
We should be using some sort of key for access for security instead of just relying on IP white listing or htaccess. Whatever the M2 way to do this is.

**Comment by julesbravo** [comment]
Hi @serzhik Which Searchspring account are you working on?

**Comment by serzhik** [comment]
@julesbravo we use Nextopia now but we need to move to SearchSpring. I'm just looking around and see that this module is implemented not in Magento way. I want to fix the main issues before we can use it.

**Comment by julesbravo** [comment]
@serzhik What domain name is your account under?

**Comment by serzhik** [comment]
> @serzhik What domain name is your account under?

gigaparts.com

**Comment by julesbravo** [comment]
@serzhik Adding a key to do this would require also re-architecting this on our indexing server side so that would not be a PR we'd be ready to accept at this time. I'll have our team review the rest of your PRs.

**Comment by serzhik** [comment]
Yes, of course. I want to prepare the code before it. It will have a switcher in Admin Panel, which behavior to use: "Restriction by IP" or "Using API Key"

**Comment by julesbravo** [comment]
@serzhik We do not have plans to support an API key on our indexing server side yet at this time. We're actually completely rewriting our M2 module now that pushes the feed to us directly instead of us accessing it from the server.

**Comment by serzhik** [comment]
@julesbravo 
I can help you if you need.

**Comment by julesbravo** [comment]
@serzhik We are working with one of our M2 agency partners on this project already.
